collector/ethtool_linux.go

```diff
@@ -116,6 +118,28 @@ func init() {
 	registerCollector("ethtool", defaultDisabled, NewEthtoolCollector)
 }
 
+// Sanitize the given metric name by replacing invalid characters by underscores.
+//
+// OpenMetrics and the Prometheus exposition format require the metric name
+// to consist only of alphanumericals and "_", ":" and they must not start
+// with digits. Since colons in MetricFamily are reserved to signal that the
+// MetricFamily is the result of a calculation or aggregation of a general
+// purpose monitoring system, colons will be replaced as well.
+//
+// Note: The caller has to ensure that the metric name does not start with a
+// digit.
```

Comment:
Why? Prometheus / OpenMetrics only places restrictions on the first character of the metric, i.e. the result of BuildFQName. A metric name of `node_ethtool_0foobar` would be valid.

Reply:
In our case we add `node_ethtool_` as prefix, but `0foobar` would be invalid.

Reply:
This would change the meaning of what this note should tell: If you do not prepend something to this sanitized name and the metric starts with a digit, the result of this function will not a valid metric name.

Reply:
Ok,
```suggestion
// Note: If not subsequently prepending a namespace and / or subsystem (e.g., 
// with prometheus.BuildFQName), the caller must ensure that the supplied
// metricName does not begin with a digit.
```

Reply:
Thanks. Updated the comment.